modules/ting_search/modules/subsearch_translate/subsearch_translate.module

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * @file
+ * Translated search.
+ */
+
+use GuzzleHttp\Client as HttpClient;
+use GuzzleHttp\Exception\ClientException as GuzzleClientException;
+
+// Load Field module hooks.
+module_load_include('inc', 'subsearch_translate', 'subsearch_translate.admin');
+
+/**
+ * Implements hook_ting_search_results().
+ */
+function subsearch_translate_ting_search_results($keys = NULL, $conditions = NULL, $results) {
+
+  $messages = array();
+
+  // If no google api key is set we cannot translate.
+  if (empty(variable_get('translated_suggestion_google_key', ''))) {
+    return $messages;
+  }
+
+  // If any of fiction or non fiction is empty we cannot do the calculation.
+  // Simply exit for now.
```

Comment:
This whole criterion seems very weird and specific to me (fiction vs non-fiction). Should we make it optional, with just a checkbox to enable it? Might make this translation subsearch feature more usable to others.